src/Database/Query.php

```diff
@@ -231,48 +237,13 @@ public function getConnection(): Connection
     }
 
     /**
-     * Sets the connection for this query to the read-only role for the current connection.
-     *
-     * @return $this
-     */
-    public function useReadRole()
```

Comment:
I get that replica usage is being confined to `SelectQuery` which is great. However, these methods are easier to use than `setConnectionRole()` is as `setConnectionRole()` requires importing Connection to get to the constants.